client/app/dedicatedCloud/name/dedicatedCloud-name.controller.js

```diff
@@ -0,0 +1,55 @@
+angular.module("App").controller("DedicatedCloudNameCtrl", class {
+
+    constructor ($scope, $stateParams, $translate, DedicatedCloud) {
+        this.$scope = $scope;
+        this.$stateParams = $stateParams;
+        this.$translate = $translate;
+        this.DedicatedCloud = DedicatedCloud;
+    }
+
+    closeModal () {
+        this.$scope.resetAction();
+    }
+
+    updateDescription () {
+        this.updating = true;
+        this.DedicatedCloud.updateDescription(this.$stateParams.productId, this.newValue);
+        this.updateName()
+            .then(() => {
+                this.$scope.setMessage(this.$translate.instant(`dedicatedCloud_edit_${this.translateWord}_success`), {
+                    type: "SUCCESS"
+                });
+                _.set(this.$scope, this.$scope.currentActionData.path, this.newValue);
+                this.closeModal();
+            }).catch((err) => {
+                this.$scope.setMessage(this.$translate.instant(`dedicatedCloud_edit_${this.translateWord}_error`, [this.newValue]), {
+                    message: err.message,
+                    type: "ERROR"
+                });
+            }).finally(() => {
+                this.updating = false;
+            });
+    }
+
+    $onInit () {
+        this.newValue = this.$scope.currentActionData.value;
+
+        switch (this.$scope.currentActionData.path) {
+        case "dedicatedCloud.description":
+            this.translateWord = "description";
```

Comment:
translateWord is a little bit too generic as a variable name, maybe you could add a function such as "getModalTitle" instead

Reply:
Ok for the variable name, but making a function could make more complex since it is used into view and controller no ?